src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxIterationCountTests.cs

```diff
@@ -0,0 +1,227 @@
+            }
+        }
+
+        [ConditionalTheory]
+        [MemberData(nameof(GetCertsWith_IterationCountExceedingDefaultLimit_MemberData))]
+        public void Import_IterationCountLimitExceeded_Throws(string name, string password, bool usesPbes2, byte[] blob, long iterationCount)
+        {
+            _ = password;
+            _ = iterationCount;
+
+            if (usesPbes2 && !PfxTests.Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            CryptographicException ce = Assert.Throws<CryptographicException>(() => Import(blob));
+            Assert.Contains("2233907", ce.Message);
+        }
+
+        [ConditionalTheory]
+        [MemberData(nameof(GetCertsWith_IterationCountExceedingDefaultLimit_MemberData))]
+        public void ImportWithPasswordOrFileName_IterationCountLimitExceeded(string name, string password, bool usesPbes2, byte[] blob, long iterationCount)
+        {
+            if (usesPbes2 && !PfxTests.Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            using (TempFileHolder tempFile = new TempFileHolder(blob))
+            {
+                string fileName = tempFile.FilePath;
+                if (PfxTests.IsPkcs12IterationCountAllowed(iterationCount, PfxTests.DefaultIterations))
+                {
+                    Assert.NotNull(Import(blob, password));
+                    Assert.NotNull(Import(blob, PfxTests.GetSecureString(password)));
+
+                    Assert.NotNull(Import(fileName));
+                    Assert.NotNull(Import(fileName, password));
+                    Assert.NotNull(Import(fileName, PfxTests.GetSecureString(password)));
+                }
+                else
+                {
+                    if (OperatingSystem.IsWindows())
+                    {
+                        // Specifying password or importing from file will still give us error because cert is beyond Windows limit.
+                        // But we will get the CryptoThrowHelper+WindowsCryptographicException.
+                        VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(blob, password));
+                        VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(blob, PfxTests.GetSecureString(password)));
+
+                        // Using a file will do as above as well.
+                        VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(fileName));
+                        VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(fileName, password));
+                        VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(fileName, PfxTests.GetSecureString(password)));
+                    }
+                }
```

Comment:
Right, it is missing the "big + non-Windows" case, which should validate that we don't throw for any of the tested methods (because they are not the passwordless import).

Reply:
Actually, we can remove the first if block since all PFXes in this test exceed the Default Iteration Count of 600k.